### PR TITLE
add incident and maintenance endpoints

### PR DESCRIPTION
**Pull request by gtabrar**
For https://github.com/muxinc/engbot/pull/34

### CODE REVIEW

File: incidents.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+type CreateIncidentRequestBody struct {
+	Incident CreateIncidentParams `json:"incident"`
+}
+
+type CreateIncidentParams struct {
+	Name           string `json:"name,omitempty"`
+	Status         string `json:"status,omitempty"`
+	ImpactOverride string `json:"impact_override,omitempty"`
+	//ScheduledFor                              time.Time         `json:"scheduled_for,omitempty"`
+	//ScheduledUntil                            time.Time         `json:"scheduled_until,omitempty"`
```
Comment (brianlieberman) [inline incidents.go:117]: is this supposed to be commented out?

File: maintenance.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+// CreateMaintenanceRequestBody is the create maintenance request body representation
+type CreateMaintenanceRequestBody struct {
+	Incident CreateMaintenanceParams `json:"incident"`
+}
+
+type CreateMaintenanceParams struct {
+	Name   string `json:"name,omitempty"`
+	Status string `json:"status,omitempty"`
+	//ImpactOverride                            string            `json:"impact_override,omitempty"`
```
Comment (brianlieberman) [inline maintenance.go:31]: this one too?

**Review by brianlieberman — APPROVED** [review]

**Comment by nagelflorian** [comment]
@gtabrar do you need any help? The linked pull request is not visible to me (maybe private?).

**Comment by gtabrar** [comment]
Hey @nagelflorian - apologies I meant to push this to a fork. I'm extending the package to include some endpoints not currently covered, such as incidents. I hope to have a PR for you to include this soon 😄

**Comment by nagelflorian** [comment]
Looking forward to your contribution! :)

**Pull request closed without merging**